### PR TITLE
modified:   source/contribute.rst to add document for local python se…

### PR DESCRIPTION
**Pull request by frog-o**
I keep looking up the command "python -m http.server 8000" to start a local web server on the web(I use a raspberry pi4 headless)..  The instructions they used don't allow for it to be severed over the LAN only localhost.  I thought that this should be part of the instructions.  This is the first time i tried to make a pull request and want to start with something simple. Please forgive me if something not correct or missing .  I did go through there forking tutorial a long while ago and it dealt with me make a pull request on my own repository nothing this advance.   

Just a little note: IMHO there a lot of needed documentation on setup tools and I am trying to do it and not just not just open a lot of issue.   I hoping to add more later and trying to get my feet wet .

### CODE REVIEW

**Comment by di** [comment]
Hi @frog-o, thanks for the contribution. I think this is probably pretty specific to your setup and probably not going to be helpful for most contributors.

One thing we could do instead is change the `--host` option for `sphinx-autobuild` from the default `127.0.0.1` to `0.0.0.0`, which would externally expose the preview for users like yourself, so you wouldn't have to run `python -m http.server 8000` anymore.

**Comment by frog-o** [comment]
I thought you had it that way for security. The only way it specifically for my setup is i am using pyenv and don't need to put 3 on front of which would make it python3 like most other distribution. Python 2 is going depreciating  soon so i figure it better left out. I thought if this was to big of an issue I could just add what i said above in clear term to the document but i thought someone might want to change the nox file instead . I figure I get feedback if nothing else it is easer for me to change the doc then the execution script i know nothing about and try to add another flag.

**Comment by di** [comment]
Yup, I hear you, and thanks for taking the time to make the doc changes. In the future if you're not sure a change will be desired or not, feel free to open an issue to ask before making the PR.

**Comment by frog-o** [comment]
Thank you I should of done that, I  always try to  think thing threw and they never go as expected.   To me it seem wrong to **make and issue and not have a fix**.  To me in a way pull request and issues are not much different but it does goes against your stander which **I need to follow I did go thought there tutorials**.   I felt since it was just such a small change it would be wrong to create an issue.  
  
Can I get a little help?

I am bit confused **what am i supposed to do now**.

Are you just some one _who committing on my pull request or a maintainer_?
  
When you say 

> One thing we could do instead is change the --host option for sphinx-autobuild from the default 127.0.0.1 to 0.0.0.0, which would externally expose the preview for users like yourself, so you wouldn't have to run python -m http.server 8000 anymore.

This would be great if that was done but it does not show _good programming practice_ which I tough you want to keep.

Or are you saying I should hit the close and comment button bellow and make an issue instead?

Thank you for take the time to answering my question an working with me this far and **I apologize for not following your standers**.

**Comment by di** [comment]
> Are you just some one who committing on my pull request or a maintainer?

I'm a maintainer, you can tell by the "Member" badge on my comments:

Screen Shot 2020-04-23 at 11 03 06 AM

> This would be great if that was done but it does not show good programming practice which I tough you want to keep.

I think this would be fine and I'd merge a PR that changes this as long as none of the other maintainers object.

> Or are you saying I should hit the close and comment button bellow and make an issue instead?

I was suggesting creating an issue in the future to confirm that the change would be desired and to avoid doing the work to make a pull request that might not get merged. 🙂 

I don't think you need to make an issue for this since we're having the discussion right here. I think you can either make a new PR that makes the change I suggested instead, or you could just do it by updating this PR if you prefer.

**Comment by frog-o** [comment]
I think i got it update took a lot of research though.

 2 hour to make the change need and 2 hours trying to figure out git. Probably going to be a hour write this Just for a change that should only take a minute.  Whew I hate being a beginner. :(

Details bellow skip if you don't want to read....

### How I spent so long 
Tried to pass a dictionary to session.run().  Nox did not like that and  I accidentally put  a space in 
 my quotes like this "0.0.0.0 " found out it did not like that. I also found out it  did not like a space between the "H " an IP address.  Each time I made a change and tested it, I had to wait for it to build the documentation.  It was very slow.    And then when i got all that working I had to look up how to revert a file and how to push those changes back to my fork etc.  

I sure the probably better way to do this.  I should of 4 example check if there was a switch to pretend build in nox.  I have a hard time read all the text in man pages though.

### Testing my changes
I hope everything ok I tried

nox -s build
nox -s preview

And load it in my web browser look like every thing good to me? But i am new and very good at doing things wrong :(

I Thankful for the time you spent with me .

File: noxfile.py [hunk 1]
```diff
@@ -16,11 +16,9 @@ def build(session, autobuild=False):
     shutil.rmtree("build", ignore_errors=True)
 
     if autobuild:
-        command = "sphinx-autobuild"
+        session.run("sphinx-autobuild","-H0.0.0.0", "-W", "-b", "html", "source", "build")
```
Comment (webknjaz) [inline noxfile.py:19]: this `-H0.0.0.0` could probably be `"-H", "0.0.0.0"`

File: noxfile.py [hunk 2]
```diff
@@ -16,11 +16,9 @@ def build(session, autobuild=False):
...
     else:
-        command = "sphinx-build"
-
-    session.run(command, "-W", "-b", "html", "source", "build")
+        session.run("sphinx-build", "-W", "-b", "html", "source", "build")
```
Comment (webknjaz) [inline noxfile.py:21]: Here's how you can do this without having to copy-paste that `session.run` multiple times:
```suggestion
        command = "sphinx-autobuild"
        extra_args = "-H", "0.0.0.0"
    else:
        command = "sphinx-build"
        extra_args = ()

    session.run(command, *extra_args, "-W", "-b", "html", "source", "build")
```

Pro tip: you can accept this suggested change by clicking the button below 👇

**Comment by frog-o** [comment]
I made the changes you ask and push it back to my fork .  when i click "resolve conversion" The web page show a arrow with a hand that say click to accept changes but nothing bellow it.  I done a bunch of google searching Trying to find how to approve change(from the submitter side) the best i found was [here](https://zellwk.com/blog/edit-pull-request/)  and it was a video which does not show much.  He went back to his fork and edit the file in github.  I believe I did the same but though the command line so i can test the code.  I could not get it to run at first because it was running in the background

it gave me this

>OSError: [Errno 98] Address already in use                            │        RX errors 0  dropped 0  overruns 0  frame 0
nox > Command sphinx-autobuild -H 0.0.0.0 -W -b html source build fail│        TX packets 5776  bytes 560297 (547.1 KiB)
>ed with exit code 1  

I saw the "-H  0.0.0.0" and thought it was trying to put the space in it. and spent some time trying to get rid of the space.  When non of that worked i decide to look at the error more closely and saw address already in use.  How do I miss such simple stuff.

**Comment by webknjaz** [comment]
> when i click "resolve conversion" The web page show a arrow with a hand that say click to accept changes but nothing bellow it

It was my mistake, the button is above, not below. It says `Commit suggestion`. Here's an article about this feature, with screenshots: https://haacked.com/archive/2019/06/03/suggested-changes/.

File: noxfile.py [hunk 3]
```diff
@@ -14,13 +14,13 @@ def build(session, autobuild=False):
     session.env["SPHINXOPTS"] = "-W"
 
     shutil.rmtree("build", ignore_errors=True)
-
```
Comment (webknjaz) [inline noxfile.py:17]: Please don't reformat unrelated parts of code, return the removed line here.

File: noxfile.py [hunk 4]
```diff
@@ -14,13 +14,13 @@ def build(session, autobuild=False):
...
     if autobuild:
         command = "sphinx-autobuild"
+        extra_args = "-H", "0.0.0.0"
     else:
         command = "sphinx-build"
-
```
Comment (webknjaz) [inline noxfile.py:22]: you removed this line, but it's better for readability when there's an empty line here separating blocks. please return it.

**Comment by webknjaz** [comment]
@frog-o also, I noticed that you have misconfigured Git on your computer and your commits don't use email confirmed in your GitHub account (when it's correct, GitHub shows your user profile avatar):
Screen Shot 2020-04-24 at 10 40 42 PM

You can fix it locally (on your machine where you run git commands) using the following commands (replace email and name with yours before executing):
```console
git config --global user.email "your.github@email.here"
git config --global user.name "Your Full Name Here"
```
after this, all new commits you make locally will have this new updated information in the committer field

**Comment by frog-o** [comment]
The guide you give me tells me i can do this by going to the files changed Tab an click where my error is and it should give me a blue box but it does not it grey out and said not supported.  I have not finish reading it however.  I just want to let you know now that I fixed my git and done the change you requested.  Thank you for letting me know of my miss-configured git and the guides. The help is really appreciated.

**Comment by webknjaz** [comment]
> it should give me a blue box but it does not it grey out and said not supported.

It's gray because it's a suggestion to the old diff that you had and you changed that part of the file by changing the same lines manually. You can only apply such suggestions if you haven't changed the lines the suggestion is modifying.

File: noxfile.py (hunk 3 above)
Comment (webknjaz) [reply noxfile.py:17]: @frog-o this line is still removed. would you mind re-adding it?

File: noxfile.py [hunk 5]
```diff
@@ -14,13 +14,14 @@ def build(session, autobuild=False):
     session.env["SPHINXOPTS"] = "-W"
 
     shutil.rmtree("build", ignore_errors=True)
-
```
Comment (webknjaz) [inline noxfile.py:17]: @frog-o Looks like you marked it as resolved but in fact, it isn't ^

**Comment by frog-o** [comment]
webknjaz you wrote
>@frog-o Looks like you marked it as resolved but in fact, it isn't ^

It is fixed, it  look to me as it is a different space then the first one though

**It was**    
>------------------------line 22 missing space ----------------------------
>20  else:
>21      command = "sphinx-build"
>22      \-

**It is now** 

>--------------------line 17 now -----------------------------------------
>14       session.env["SPHINXOPTS"] = "-W"
>15
>16   shutil.rmtree("build", ignore_errors=True)
>17 \-

I look over the file on github just now and compared it side by side; Then i  put extra space in it where it appeared to be missing and ran  

>git diff -a master cont-python noxfile.py  

I am trying to make sure I got all of the spaces put back this time.  This is the output:
<pre>
      session.env["SPHINXOPTS"] = "-W" 

      shutil.rmtree("build", ignore_errors=True)               
-                                                             
+     
       if autobuild:                                            
             command = "sphinx-autobuild"                         
+            extra_args = "-H", "0.0.0.0"                         
        else:                                                    
            command = "sphinx-build"                             
+           extra_args = ()    

-      session.run(command, "-W", "-b", "html", "source", "build│l si") 
+      session.run(command, *extra_args, "-W", "-b", "html", "so│o 80urce", "build") 
</pre>

It show one extra line which i put in to match what GitHub was showing me when i view it.

frog-o
> it should give me a blue box but it does not it grey out and said not supported.

----

webknjaz 
>It's gray because it's a suggestion to the old diff that you had and you changed that part of the file >by changing the same lines manually. You can only apply such suggestions if you haven't changed >the lines the suggestion is modifying.

I tried it before i made the commit the diff should still be good and tried again after you posted this still no luck.  I am **guessing**  it only work, if I was not  the _one_ make the pull request, since i made the pull request and have a **fork** it does not work.  My understanding of that post is to help people who helping review the code(who not necessarily a maintainer or have write access) to commit changes without *forking*.

If you have anymore problems I suggest you just merge it and put the space in it yourself since I can not do it right sorry about that :(.

Note: I would love to give you credit for help me but I forget howto.  I know there a link to it on github if I can just find it.

File: noxfile.py [hunk 6]
```diff
@@ -14,13 +14,15 @@ def build(session, autobuild=False):
...
+
```
Comment (webknjaz) [inline noxfile.py:17]: It looks like you accidentally added 4 extra spaces here. Let's remove them to get the original line as it was before:
```suggestion

```
Do you want to try practicing that `Commit suggestion`  ☝️ button? (this only removes four spaces but keeps the empty line in place.

**Comment by webknjaz** [comment]
> since i made the pull request and have a **fork** it does not work

No, it works perfectly forks. Try the new suggestion that I just posted.

**Comment by webknjaz** [comment]
> Note: I would love to give you credit for help me but I forget howto. I know there a link to it on github if I can just find it.

This is possible with `Co-authored-by` commit trailers but nevermind. Let's Just get it done :)

> My understanding of that post is to help people who helping review the code(who not necessarily a maintainer or have write access) to commit changes without forking.

People with write access to this repository (maintainers) have access to the PR branch in your fork (`cont-python` in this case) unless you forbid it by unchecking the corresponding checkbox (https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

File: noxfile.py [hunk 7]
```diff
@@ -14,13 +14,14 @@ def build(session, autobuild=False):
     session.env["SPHINXOPTS"] = "-W"
 
     shutil.rmtree("build", ignore_errors=True)
```
Comment (webknjaz) [inline noxfile.py:16]: I'm confused why that patch didn't recover the line. Let's try like this:
```suggestion
    shutil.rmtree("build", ignore_errors=True)

```

**Comment by webknjaz** [comment]
@di this should be ready now

**Pull request merged**